src/jit/codegenxarch.cpp

```diff
@@ -1461,63 +1461,50 @@ CodeGen::genStructReturn(GenTreePtr treeNode)
     var_types targetType = treeNode->TypeGet();
 
 #ifdef FEATURE_UNIX_AMD64_STRUCT_PASSING
-    if (targetType == TYP_VOID)
-    {
-        assert(op1 == nullptr);
-        if (compiler->info.compRetBuffArg != BAD_VAR_NUM)
-        {
-            // System V AMD64 spec requires that when a struct is returned by a hidden
-            // argument the RAX should contain the value of the hidden retbuf arg.
-            getEmitter()->emitIns_R_S(INS_mov, EA_BYREF, REG_RAX, compiler->info.compRetBuffArg, 0);
-        }        
-    }
-    else
-    {
-        noway_assert((op1->OperGet() == GT_LCL_VAR) ||
-                     (op1->OperGet() == GT_CALL));
+    noway_assert((op1->OperGet() == GT_LCL_VAR) ||
```

Comment:
Even isStructReturn() needs modifications.  Earlier it was returning true for RetBuf returning methods since it wasn't properly modelled in IR.  Since that is addressed, lines 1439 to 1440 could be replaced with 

return varTypeIsStruct(treeNode);